File: src/main/java/at/pria/joust/webapp/web/joust/TeamsController.java
package at.pria.joust.webapp.web.joust;

import at.pria.joust.Team;
import at.pria.joust.Tournament;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Controller;
import org.springframework.ui.Model;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;

@Controller
public class TeamsController {
    @Autowired
    private Tournament tournament;

    @RequestMapping(value = "/admin/teams/", method = RequestMethod.GET)
    public String adminTeams(Model model) {
        model.addAttribute("tournament", tournament);
        model.addAttribute(new SeedingInput());
        return "joust/teams_admin";
    }

    @RequestMapping(value = "/admin/teams/", method = RequestMethod.POST)
    public String adminTeamsPost(Model model, SeedingInput seedingInput) {
        seedingInput.apply(tournament);
        return adminTeams(model);
    }

    @RequestMapping(value = "/teams/", method = RequestMethod.GET)
    public String teams(Model model) {
        model.addAttribute("tournament", tournament);
        return "joust/teams";
    }

    private static class SeedingInput {
        private String team;
        private int round;
        private int score;

        public void apply(Tournament t) {
            for (Team team : t.getTeams())
                if (team.getId().equals(this.team)) {
                    t.getSeedingResults().setResult(team, round, score);
                    break;
                }
        }

        public String getTeam() {
            return team;
        }

        public void setTeam(String team) {
            this.team = team;
        }

        public int getRound() {
            return round;
        }

        public void setRound(int round) {
            this.round = round;
        }

        public int getScore() {
            return score;
        }

        public void setScore(int score) {
            this.score = score;
        }
    }
}

File: src/main/scala/at/pria/joust/webapp/web/joust/TeamsController.scala
package at.pria.joust.webapp.web.joust

import scala.beans.BeanProperty

import at.pria.joust._

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Controller
import org.springframework.ui.Model
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestMethod

@Controller
class TeamsController {
  @Autowired
  private[this] var tournament: Tournament = _

  @RequestMapping(value = Array("/admin/teams/"), method = Array(RequestMethod.GET))
  def adminTeams(model: Model) = {
    model.addAttribute("tournament", tournament)
    model.addAttribute(new TeamsControllerInput())
    "joust/teams_admin"
  }

  @RequestMapping(value = Array("/admin/teams/"), method = Array(RequestMethod.POST))
  def adminTeamsPost(model: Model, teamsControllerInput: TeamsControllerInput) = {
    teamsControllerInput.apply(tournament)
    adminTeams(model)
  }

  @RequestMapping(value = Array("/teams/"), method = Array(RequestMethod.GET))
  def teams(model: Model) = {
    model.addAttribute("tournament", tournament)
    "joust/teams"
  }
}

class TeamsControllerInput {
  @BeanProperty var team: String = _
  @BeanProperty var round: Int = _
  @BeanProperty var score: Int = _

  def apply(t: Tournament) = {
    val team = t.teams.find { _.id == this.team }.get
    t.seedingResults.result(team, round, score)
  }
}
